charon/src/bin/charon-driver/translate/translate_trait_objects.rs

```diff
@@ -679,16 +744,98 @@ impl ItemTransCtx<'_, '_> {
                     });
                     ConstantExprKind::Ref(global)
                 }
-                // TODO(dyn): builtin impls
-                _ => ConstantExprKind::Opaque("missing supertrait vtable".into()),
+                hax::ImplExprAtom::Builtin { trait_data, .. } => {
+                    // Handle built-in implementations, including closures
+                    let tref = &impl_expr.r#trait;
+                    let hax_state = self.hax_state_with_id();
+                    let trait_def = self.hax_def(&tref.hax_skip_binder_ref().erase(hax_state))?;
+
+                    trace!(
+                        "Processing builtin impl for trait {:?}, lang_item: {:?}, trait_data: {:?}",
+                        trait_def.def_id(),
+                        trait_def.lang_item,
+                        trait_data
+                    );
+
+                    let closure_kind = trait_def.lang_item.as_deref().and_then(|lang| match lang {
+                        "fn_once" => Some(ClosureKind::FnOnce),
+                        "fn_mut" => Some(ClosureKind::FnMut),
+                        "fn" | "r#fn" => Some(ClosureKind::Fn),
+                        _ => None,
+                    });
+
+                    // Check if this is a closure trait implementation
+                    if let Some(closure_kind) = closure_kind
+                        && let Some(hax::GenericArg::Type(closure_ty)) =
+                            impl_expr.r#trait.hax_skip_binder_ref().generic_args.first()
+                        && let hax::TyKind::Closure(closure_args) = closure_ty.kind()
+                    {
+                        // Register the closure vtable instance
+                        let vtable_instance_ref: GlobalDeclRef = self.translate_item(
+                            span,
+                            &closure_args.item,
+                            TransItemSourceKind::VTableInstance(TraitImplSource::Closure(
+                                closure_kind,
+                            )),
+                        )?;
+                        let global = Box::new(ConstantExpr {
+                            kind: ConstantExprKind::Global(vtable_instance_ref),
+                            ty: fn_ptr_ty,
+                        });
+                        ConstantExprKind::Ref(global)
+                    } else if self
+                        .translate_vtable_struct_ref(span, impl_expr.r#trait.hax_skip_binder_ref())?
```

Comment:
You can't skip binders like that, this will result in errors if the binder is not trivial.. Please use `translate_region_binder` then erase the resulting `charon::RegionBinder`.